pkgs/applications/science/molecular-dynamics/lammps/default.nix

```diff
@@ -0,0 +1,52 @@
+{ stdenv, writeText, fetchurl,
+  libpng, fftw,
+  mpiSupport ? false, mpi ? null
+}:
+
+assert mpiSupport -> mpi != null;
+
+stdenv.mkDerivation rec {
+  version = "16Feb16";
+  name = "lammps-${version}";
+
+  src = fetchurl {
+    url = "http://downloads.sourceforge.net/project/lammps/${ name }.tar.gz";
+    sha256 = "058459f3053cca0b8c0fdd05cca7acab6ac00934c6da8c9008aa0f55fb5ceefb";
+  };
+
+  passthru = {
+    inherit mpi;
+  };
+
+  buildInputs = [ fftw libpng ]
+  ++ (stdenv.lib.optionals mpiSupport [ mpi ]);
+
+  builder = writeText "builder.sh" ''
```

Comment:
Yes there is a reason I am not using the standard phase hook... It requires two separate makes to create the executable and libraries. 

Is there a way to run make twice with different arguments?

For example:
make mode=shlib mpi
make mode=exe mpi